frontend/tests/console/utils/DebugBundlePage.ts

```diff
@@ -0,0 +1,357 @@
+      this.page.getByText(/download/i).waitFor({ timeout }),
+      this.page.getByText(/error|failed/i).waitFor({ timeout }),
+    ]);
+  }
+
+  /**
+   * Validation methods
+   */
+  async verifyBrokerStatus() {
+    await expect(this.page).toHaveURL(/\/debug-bundle\/progress\//);
+
+    // Verify the overview component is visible
+    const overview = this.page.getByTestId('debug-bundle-overview');
+    await expect(overview).toBeVisible({ timeout: 5000 });
+
+    // Verify at least one broker status is visible
+    const brokerStatus = this.page.locator('[data-testid^="debug-bundle-broker-status-"]').first();
+    await expect(brokerStatus).toBeVisible({ timeout: 5000 });
+  }
+
+  async verifyGenerationInProgress() {
+    const generatingText = this.page.getByTestId('debug-bundle-generating-text');
+    await expect(generatingText).toBeVisible();
+    await expect(generatingText).toHaveText('Generating bundle...');
+  }
+
+  async verifyGenerationComplete() {
+    const completeBox = this.page.getByTestId('debug-bundle-complete-box');
+    await expect(completeBox).toBeVisible({ timeout: 60_000 });
+  }
+
+  async verifyGenerationFailed() {
+    const errorText = this.page.getByTestId('debug-bundle-error-text');
+    await expect(errorText).toBeVisible();
+    await expect(errorText).toHaveText('Your debug bundle was not generated.');
+  }
+
+  async verifyBundleExpired() {
+    const expiredText = this.page.getByTestId('debug-bundle-expired-text');
+    await expect(expiredText).toBeVisible();
+    await expect(expiredText).toHaveText('Your previous bundle has expired and cannot be downloaded.');
+  }
+
+  async verifyAdvancedModeActive() {
+    await expect(
+      this.page.getByText(
+        /This is an advanced feature, best used if you have received direction to do so from Redpanda support./i
+      )
+    ).toBeVisible();
+  }
+
+  async verifyBasicModeActive() {
+    const generateButton = this.page.getByRole('button', { name: /generate/i }).first();
+    await expect(generateButton).toBeVisible();
+  }
+
+  /**
+   * Verify specific broker status
+   */
+  async verifyBrokerStatusById(brokerId: number) {
+    const brokerStatus = this.page.getByTestId(`debug-bundle-broker-status-${brokerId}`);
+    await expect(brokerStatus).toBeVisible();
+  }
+
+  async verifyBrokerLabel(brokerId: number) {
+    const brokerLabel = this.page.getByTestId(`broker-${brokerId}-label`);
+    await expect(brokerLabel).toBeVisible();
+    await expect(brokerLabel).toHaveText(`Broker ${brokerId}`);
+  }
+
+  async verifyBrokerError(brokerId: number, expectedMessage?: string) {
+    const errorLabel = this.page.getByTestId(`broker-${brokerId}-error-label`);
+    await expect(errorLabel).toBeVisible();
+
+    if (expectedMessage) {
+      const errorMessage = this.page.getByTestId(`broker-${brokerId}-error-message`);
+      await expect(errorMessage).toHaveText(expectedMessage);
+    }
+  }
```

Comment:
I would love it if each of these self-contained functions could be wrapped in a `test.step` at some level, could be within each function or elsewhere.